### DISCUSSION TITLE
Thermostat using secondary temperature sensor

### DISCUSSION THREAD

**Discussion by dsbeach** [HomeSpan/HomeSpan#388] [Ideas]
I need some help getting started with my thermostat project.  I'd like to create a thermostat whose behavior is dependent on the current reading of a secondary temperature sensor. The idea is to control a duct fan to redirect heat from upper level of an airplane hangar down to a lower level.  If the thermostat (on the lower level) is set to 68F, and the upper level temperature sensor indicates warm air is available, then it should turn on the duct fan until the desired temperature is reached at the lower level.

I've found good sample code for the temp sensor and thermostat, but how should the thermostat read the secondary temp sensor?  Any guidance would be appreciated.

**Comment by HomeSpan** [comment]
This is a timely question.  In principle you can create two separate HomeSpan devices - one is a thermostat with temp sensor and the other is just a temp sensor - and create logic within the Home App to control the thermostat device based on the remote temp sensor.  However, the Home App has very limited automation features and I'm not sure it is possible to create such a complex series of operations.

To address this issue more broadly, I've been working on adding native support for ESP-NOW to HomeSpan.  I'm designing this for the express purpose of adding multiple remote sensors to send readings back to HomeSpan in an integrated fashion.  I'm about half-way through the design and development - it will be the basis for the next release.  If it works as intended, the remote sensors will be able to operate for an extended period of time on just battery, making it easy to place anywhere.

I'm probably about 6 weeks out from the next release.  In the interim may want to implement a standard HomeSpan thermostat device to make sure it is all working.

**Reply by dsbeach** [reply]
I've got plenty of work to do on my thermostat code. I'll happily focus on it's logic and 'stub out' the remote sensor read for now.

Great news, thanks for all you do for the HomeSpan library.

P.S. As a retired software developer, I'm happy to beta test when the time comes.
